compiler/rustc_trait_selection/src/solve/fulfill.rs

```diff
@@ -197,6 +197,12 @@ where
                     delegate.compute_goal_fast_path(goal, obligation.cause.span)
                 {
                     match certainty {
+                        // Even if the goal references two instances of the same infer var,
+                        // our fast paths should not depend on region identity.
+                        //
+                        // Only goals proven via the trait solver should be region dependent.
+                        // We never call `push_hir_typeck_potentially_region_dependent_goal`
+                        // here.
```

Comment:
```suggestion
                        // This fast path doesn't depend on region identity so it doesn't
                        // matter if the goal contains inference variables or not, so we
                        // don't need to call `push_hir_typeck_potentially_region_dependent_goal`
                        // here.
                        //
                        // Only goals proven via the trait solver should be region dependent.
```
I found this comment confusing the first time round as I read it as "we need to prevent all fast paths from caring about region identity". Whereas this seems to be saying "*this* fast path already doesn't care about region identity so we don't need to prevent it from caring by calling this method".

It also feels funny to call out "two instances of the same infer" when we later care about *all* infers not just *repeated* ones :thinking: